crates/rpc-trace-types/src/opcode.rs

```diff
@@ -0,0 +1,42 @@
+//! Types for opcode tracing.
+
+use alloy_primitives::B256;
+use serde::{Deserialize, Serialize};
+
+/// Opcode gas usage for a transaction.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct BlockOpcodeGas {
+    /// The block hash
+    pub block_hash: B256,
+    /// The block number
+    pub block_number: u64,
+    /// All executed transactions in the block in the order they were executed, with their opcode
+    /// gas usage.
+    pub transactions: Vec<TransactionOpcodeGas>,
+}
+
+/// Opcode gas usage for a transaction.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TransactionOpcodeGas {
+    /// The transaction hash
+    pub transaction_hash: B256,
+    /// The gas used by each opcode in the transaction
+    pub opcode_gas: Vec<OpcodeGas>,
+}
+
+/// Gas information for a single opcode.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct OpcodeGas {
+    /// The name of the opcode
+    pub opcode: String,
+    /// How many times the opcode was executed
+    pub count: u64,
+    /// Combined gas used by all instances of the opcode
+    ///
+    /// For dynamic gas costs, this is the sum of all gas used by the opcode.
```

Comment:
this first sentence is true of both dynamic and constant gas costs

so maybe remove the "For dynamic gas costs" phrasing